dimod/binary/binary_quadratic_model.py

```diff
@@ -758,6 +758,93 @@ def add_linear_equality_constraint(
                     u, v, 2 * lagrange_multiplier * ubias * vbias)
         self.offset += lagrange_multiplier * constant * constant
 
+    def add_linear_inequality_constraint(
+            self, terms: Iterable[Tuple[Variable, Bias]],
+            lagrange_multiplier: Bias,
+            label: str,
+            constant: int = 0,
+            lb: int = - np.Infinity,
+            ub: int = 0,
+            slack_method: str = "log2",
+            cross_zero: bool = False
+    ):
+        """Add a linear inequality constraint as a quadratic objective.
+
+        Adds a linear inequality constraint of the form:
+
+        math:'lb <= \sum_{i,k} a_{i,k} x_{i,k} + constant <= ub'
+        to the binary quadratic model as a quadratic objective.
+        For constraints with fractional coefficients, multiply both sides of the inequality by an
+        appropriate factor of ten to attain or approximate integer coefficients.
+
+        Args:
+            terms (iterable/iterator):
+                An iterable of 2-tuples, (variable, bias).
+                Each tuple is evaluated to the term (bias * variable).
+                All terms in the list are summed.
+            lagrange_multiplier:
+                The coefficient or the penalty strength. This value is
+                multiplied by the entire constraint objective and added to the
+                bqm (it doesn't appear explicitly in the equation above).
+            label:
+                Prefix used to label the slack variables used to create the new objective.
+            constant:
+                The constant value of the constraint.
+            lb:
+                lower bound for the constraint
+            ub:
+                upper bound for the constraint
+            slack_method:
+                "The method for adding slack variables. Supported methods are:
+                - log2: Adds log2(slack_range) number of binary variables to the constraint.
+            cross_zero
+                If True, includes zero as the domain of constraint if not include already
+
+        """
+        assert (slack_method in ['log2'])
+
+        terms_upper_bound = sum(v for _, v in terms if v > 0)
+        terms_lower_bound = sum(v for _, v in terms if v < 0)
+        ub_c = min(terms_upper_bound, ub - constant)
+        lb_c = max(terms_lower_bound, lb - constant)
+        slack_terms = []
+
+        if terms_upper_bound <= ub_c and terms_lower_bound >= lb_c:
+            print(f'Did not add constraint {label}. This constraint is feasible with any value for state variables.')
+            return slack_terms
+
+        if ub_c <= lb_c:
+            print(f'Did not add constraint {label}. This constraint is infeasible with any value for state variables.')
+            return slack_terms
+
+        slack_upper_bound = int(ub_c - lb_c)
+        if slack_upper_bound == 0:
+            print(f'add constraint {label} as equality constraint.')
```

Comment:
Either raise an error telling the user to use `add_linear_equality_constraint` or just accept it quietly without the print statement.

Reply:
I'll go with accept it quietly without print as user might not be able to define if a constraint binds.